controllers/storagecluster/external_resources.go

```diff
@@ -326,6 +339,108 @@ func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, inst
 	return reconcile.Result{}, nil
 }
 
+// claims should be created only once and should not be created/updated again if user deletes/update it.
+func (r *StorageClusterReconciler) createDefaultStorageClassClaims(instance *ocsv1.StorageCluster) error {
+
+	StorageClassClaimFile := &ocsv1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForCephFilesystemSC(instance),
+			Namespace: instance.Namespace,
+			Labels: map[string]string{
+				defaultStorageClassClaimLabel: "true",
+			},
+		},
+		Spec: ocsv1alpha1.StorageClassClaimSpec{
+			Type: "sharedfilesystem",
+		},
+	}
+
+	StorageClassClaimBlock := &ocsv1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForCephBlockPoolSC(instance),
+			Namespace: instance.Namespace,
+			Labels: map[string]string{
+				defaultStorageClassClaimLabel: "true",
+			},
+		},
+		Spec: ocsv1alpha1.StorageClassClaimSpec{
+			Type: "blockpool",
+		},
+	}
+
+	err := r.createAndOwnStorageClassClaim(instance, StorageClassClaimFile)
+	if err != nil {
+		return err
+	}
+
+	err = r.createAndOwnStorageClassClaim(instance, StorageClassClaimBlock)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *StorageClusterReconciler) createAndOwnStorageClassClaim(
+	instance *ocsv1.StorageCluster, claim *ocsv1alpha1.StorageClassClaim) error {
+
+	err := controllerutil.SetOwnerReference(instance, claim, r.Client.Scheme())
+	if err != nil {
+		return err
+	}
+
+	err = r.Client.Create(context.TODO(), claim)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
```

Comment:
Since we want default claims to be created only once, isn't it a bigger issue if there is an existing one with same name? We should probably error out when we think we need to create default claim but for some reason it already exists?

Reply:
Actually, this is a protection against failure in the creation of a second claim. 

For example, if we fail to create a second one we will return an error which means we will call the same code again to create the second one and the first one will be already existing. If we do not add this condition and fail on it again and again we won't be able to proceed with reconciliation at all.

Reply:
@iamniting Please undo as of comments from last reivew

Reply:
In that case, how do you know if the existing claim is the expected default?

Reply:
1. This code is only called during onboarding
2. If the customer somehow managed to create an scc with the same name before onboarding then we would like to skip that one, and continue with the creation of the second one

Reply:
>     2. If the customer somehow managed to create an scc with the same name before onboarding then we would like to skip that one, and continue with the creation of the second one
>
But how do we skip that and create another one with consistent naming? We can not use a different name whenever there is an AlreadyExists error. If we don't return an AlreadyExists error, we won't even know about it.

Reply:
> But how do we skip that and create another one with consistent naming?
This is not the use case we are trying to solve. We will not create another one to replace the one the user created. If a customer creates a stoageclassclaim with the same name we assume he knows what he is doing. 

But if the customer only create one of the 2 we will create the second one.